recipes/aws-cdi-sdk/all/conanfile.py

```diff
@@ -1,144 +1,121 @@
+        deps.set_property("aws-sdk-cpp", "cmake_target_name", "aws-cpp-sdk-core")
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
-        autotools = self._configure_autotools()
-        with tools.chdir(self._source_subfolder):
-            # configure autotools to find aws-cpp-sdk-cdi
-            autotools.include_paths.append(os.path.join(self.build_folder, self._source_subfolder, "aws-cpp-sdk-cdi", "include"))
-            autotools.library_paths.append(os.path.join(self.build_folder, "lib"))
-            autotools.libs.append("aws-cpp-sdk-cdi")
-
-            vars = autotools.vars
-            cc, cxx = self._detect_compilers()
-            vars["CC"] = cc
-            vars["CXX"] = cxx
-            if self.settings.build_type == "Debug":
-                vars["DEBUG"] = "y"
-
-            args = ["require_aws_sdk=no"]
-
-            autotools.make(target="libsdk", vars=vars, args=args)
-
     def package(self):
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "include"))
-        config = "debug" if self.settings.build_type == "Debug" else "release"
-        self.copy(pattern="*", dst="lib", src=os.path.join(self._source_subfolder, "build", config, "lib"))
-
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-
-    def package_info(self):        
+    def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "aws-cdi-sdk")
-        
+        self.cpp_info.set_property("cmake_target_aliases", ["aws-cdi-sdk"])
+
+        cpp_sdk = self.cpp_info.components["aws-cpp-sdk-cdi"]
+        cpp_sdk.set_property("cmake_target_name", "AWS::aws-cpp-sdk-cdi")
+        cpp_sdk.set_property("pkg_config_name", "aws-cpp-sdk-cdi")
+        cpp_sdk.libs = ["aws-cpp-sdk-cdi"]
+        cpp_sdk.requires = ["aws-sdk-cpp::monitoring", "aws-libfabric::aws-libfabric"]
+
+        c_sdk = self.cpp_info.components["cdisdk"]
+        c_sdk.set_property("cmake_target_name", "AWS::aws-cdi-sdk")
+        c_sdk.set_property("pkg_config_name", "aws-cdi-sdk")
+        c_sdk.libs = ["cdisdk"]
+        c_sdk.requires = ["aws-cpp-sdk-cdi"]
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            c_sdk.defines = ["_LINUX"]
+        libcxx = stdcpp_library(self)
+        if libcxx:
+            c_sdk.system_libs.append(libcxx)
+
         # TODO: to remove in conan v2 once cmake_find_package_* generators removed
```

Comment:
As you are dropping test_v1_package, you could also remove conan v1 related stuff!

Reply:
Does Conan v2 automatically add libcxx to the linker flags? Is this mentioned somewhere in the docs as well?

Reply:
@perseoGI  - please remember that in line with the announcement https://github.com/conan-io/conan-center-index/discussions/25461 and previous discussions - we are asking contributors to not remove old logic, even when it seems trivial - literally every PR that is currently open and to be opened could have something like this - and we want to ensure PRs are as minimal as possible to make them easier to review, given the backlog. 

We also don't want to burden contributors with maintenance that we can do ourselves.  - the same goes with `test_v1_package` which we will remove at a later date all at once. 

As a side note: a lot of those "remove once the recipe is conan v2 only" are actually wrong and misleading - I've already seen some recipes were there's lines that _are_ needed for conan 2, right below that comment - so it's not as clear cut as it seems to be - and it's probably better to have automation for this to do it as safely as possible.  Also, I'd say we don't want a contributor to turn a perfectly valid and CI-passing PR into something that is broken because they removed the wrong line on our instruction.